.github/workflows/ci-tests.yml

```
@@ -0,0 +1,240 @@
+          while read -r cmd
+          do
+            eval sudo $cmd
+          done < <(Rscript -e 'writeLines(remotes::system_requirements("${{ matrix.config.os-name }}", "${{ matrix.config.os-version }}"))')
+
+      - name: Install dependencies
+        run: |
+          remotes::install_cran("covr")
+          remotes::install_deps(dependencies = TRUE, repos=options('repos')[['repos']][['RSPM']])
+          remotes::install_cran("rcmdcheck")
+        shell: Rscript {0}
+
+      - name: Session info
+        run: |
+          options(width = 100)
+          pkgs <- installed.packages()[, "Package"]
+          sessionInfo()
+        shell: Rscript {0}
+
+      - name: Check without doc
+        if: ${{!matrix.config.vignettes}}
+        env:
+          _R_CHECK_CRAN_INCOMING_: false
+          FULL_TEST_SUITE: 1
+        run: |-
+            check_args <- if("${{matrix.config.os-name}}" == "windows") {
+              c("--no-manual", "--no-vignettes", "--no-build-vignettes", "--check-subdirs=no", "--no-multiarch")
+            } else {
+              c("--no-manual", "--no-vignettes", "--no-build-vignettes", "--check-subdirs=no")
+            }
+            rcmdcheck::rcmdcheck(args = check_args, build_args=c("--no-build-vignettes"), error_on = "error", check_dir = "check")
+        shell: Rscript {0}
+
+      - name: Check with doc
+        if: matrix.config.vignettes
+        env:
+          _R_CHECK_CRAN_INCOMING_: false
+          FULL_TEST_SUITE: 1
+        run: >- 
+          rcmdcheck::rcmdcheck(args = c("--as-cran"), 
+          build_args=c("--compact-vignettes=gs+qpdf"), error_on = "warning", check_dir = "check")
+        shell: Rscript {0}
+
+      - name: run complete test suite # some tests are not run in r check, as test xls(x) files are Rbuild-ignored
+        env:
+          _R_CHECK_CRAN_INCOMING_: false
+          FULL_TEST_SUITE: 1
+          RCMDCHECK: "FALSE"
+        run: |
+         install_opts = if("${{matrix.config.os-name}}" == "windows") {
+           c("--no-multiarch")
+         } else {
+           c()
+         }
+         install.packages(list.files("./check", pattern = "XLConnect_.*.tar.gz", full.names = TRUE), repos=NULL, type="source", INSTALL_opts=install_opts)
+         setwd("tests")
+         source("run_tests.R")
+        shell: Rscript {0}
+
+      - name: Show test output
+        if: always()
+        run: | 
+          cat tests/XLConnect_Unit_Tests.txt || true
+          git clean -fd "*xls*"
+          grep -i "0 errors, 0 failures" tests/XLConnect_Unit_Tests.txt
+        shell: bash
+
+      - name: coverage report
+        if: success()
+        run: |
+          library(covr)
+          covr::codecov(function_exclusions=c("onLoad", "onAttach", "xlcEnsureDependencies"))
+        shell: Rscript {0}
+      
+      - name: Upload check results
+        if: failure()
+        uses: actions/upload-artifact@main
+        with:
+          name: ${{ runner.os }}-r${{ matrix.config.r }}-results
```

Besides the fix `r`=>`r-version`, should this be as granular as what defines the matrix runs? In case of failures for different runs we want to make sure there are different artifacts
```suggestion
          name: ${{ runner.os }}-r${{ matrix.config.r-version }}-results
```